fuse_ll/src/fuse/mount.rs

```diff
@@ -8,23 +8,126 @@ use std::os::raw::c_void;
 use std::os::unix::io::RawFd;
 use std::path::Path;
 
+use common_param::*;
 use param::*;
 
+/// Mount option validator function
+pub fn options_validator(option: String) -> Result<(), String> {
+    let ops = option.split(',');
+    for op in ops {
+        if !VALID_ON_OFF_OPTIONS.contains(&op) && !key_value_validator(&op) {
+            return Err(format!("Invalid option \"{}\", valid options: ", op)
+                + &VALID_ON_OFF_OPTIONS.join(",")
+                + ","
+                + &VALID_KEY_VALUE_OPTIONS.join(","));
+        }
+    }
+    Ok(())
+}
+
+fn key_value_validator(option: &str) -> bool {
+    match option.rfind('=') {
+        Some(a) => {
+            if a == option.len() - 1 {
+                return false;
+            }
+        }
+        None => return false,
+    }
+    for opt in VALID_KEY_VALUE_OPTIONS.iter() {
+        if match_key_value(opt, option) {
+            return true;
+        }
+    }
+    false
+}
+
+mod common_param {
+    pub const ALLOW_OTHER: &str = "allow_other";
+    pub const RW: &str = "rw";
+    pub const RO: &str = "ro";
+    pub const FSNAME: &str = "fsname=<name>";
+
+    pub fn match_key_value(pattern: &str, option: &str) -> bool {
+        pattern.split('=').nth(0) == option.split('=').nth(0)
+    }
+
+    pub fn get_value(option: &str) -> String {
+        String::from(option.split('=').last().unwrap()) //Safe to use unwrap here, becuase option is always valid.
```

Comment:
// safe to use unwrap here.
 ^ missing a space, may need cargo fmt before submitting PR